src/main/java/org/checkerframework/specimin/StandardTypeRuleDependencyMap.java

```diff
@@ -609,7 +632,37 @@ private void getAllMustImplementMethodsImpl(
           isInterface = true;
           methods = reflectionInterfaceDeclaration.getDeclaredMethods();
         } else {
-          continue;
+          methods = Set.of();
+        }
+
+        // Compose previousTypeParametersMap with the current typeParametersMap.
+        // For example, if previousTypeParametersMap is T --> String and current is E --> T,
+        // then the composed map should be E --> String.
+        List<Pair<ResolvedTypeParameterDeclaration, ResolvedType>> typeParametersMap;
+
+        // Now, substitute previous type parameters through the current map
```

Comment:
Do we have a test case that requires this behavior? If not, we should write one to double check that this is correct. (I recall that this wasn't in the last commit, so if adding it caused a test to go from failing to passing, we're already ok. If not, though, you should write a corresponding test.)

Reply:
Nevermind, I found MustImplementMethodsComplexTest